.github/workflows/build-release.yml

```diff
@@ -0,0 +1,30 @@
+name: build release
+
+on:
+  release:
+    types: [created]
+
+jobs:
+  build-release:
+    name: Release ${{ matrix.target }}
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - target: x86_64-pc-windows-gnu
+            archive: zip
+          - target: x86_64-unknown-linux-musl
+            archive: tar.gz
+          - target: x86_64-apple-darwin
+            archive: zip
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          ref: "main"
```

Comment:
Actually, this might not even be needed based on the documentation (https://github.com/actions/checkout):
> Otherwise, uses the default branch.